### ISSUE TITLE
Crash upon folder creation if already exists

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#2330]
During generation of QC report, a folder under _asset/img is created. If it already exists, the program crashes. This happens when processing several subjects in parallel (e.g. with GNU parallel), and some permission issues are encountered upon folder creation: I suspect two jobs are trying to create the folder at the same time.

<details><summary>Example of crash</summary>

~~~
Traceback (most recent call last):
  File "/Users/julien/code/sct/scripts/sct_deepseg_sc.py", line 185, in <module>
    main()
  File "/Users/julien/code/sct/scripts/sct_deepseg_sc.py", line 180, in main
    dataset=qc_dataset, subject=qc_subject, process='sct_deepseg_sc')
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 593, in generate_qc
    stretch_contrast_method='equalized',
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 491, in add_entry
    layout(qcslice)
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 281, in wrapped_f
    self.qc_report.update_description_file(img.shape)
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 432, in update_description_file
    self._update_html_assets(get_json_data_from_path(path_json))
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 448, in _update_html_assets
    os.makedirs(dest_full_path)
  File "/Users/julien/code/sct/python/envs/venv_sct/lib/python3.6/os.py", line 220, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/Users/julien/data/spineGeneric_sub/qc/_assets/imgs'
~~~

</details>

A similar (although different) issue was encountered in the past, when trying to write into the same json file, see https://github.com/neuropoly/spinalcordtoolbox/issues/2248

**Comment by jcohenadad** [comment]
Cannot reproduce

**Comment by jcohenadad** [comment]
can (and did) reproduce, on SCT course London 2020, for multi-subject processing:
~~~
*** Generate Quality Control (QC) html report ***
Resample images to 0.6x0.6 mm
QcImage: layout with Axial slice
Compute center of mass at each slice

cp /Users/julien/code/sct/spinalcordtoolbox/reports/assets/_assets/css/bootstrap.min.css /Users/julien/Desktop/sct_course_london20/multi_subject/results/qc/_assets/css
cp /Users/julien/code/sct/spinalcordtoolbox/reports/assets/_assets/css/select2.min.css /Users/julien/Desktop/sct_course_london20/multi_subject/results/qc/_assets/css
cp /Users/julien/code/sct/spinalcordtoolbox/reports/assets/_assets/css/bootstrap.min.css.map /Users/julien/Desktop/sct_course_london20/multi_subject/results/qc/_assets/css
cp /Users/julien/code/sct/spinalcordtoolbox/reports/assets/_assets/js/jquery-3.1.0.min.js /Users/julien/Desktop/sct_course_london20/multi_subject/results/qc/_assets/js
cp /Users/julien/code/sct/spinalcordtoolbox/reports/assets/_assets/js/bootstrap-table.min.js /Users/julien/Desktop/sct_course_london20/multi_subject/results/qc/_assets/js
cp /Users/julien/code/sct/spinalcordtoolbox/reports/assets/_assets/js/main.js /Users/julien/Desktop/sct_course_london20/multi_subject/results/qc/_assets/js
cp /Users/julien/code/sct/spinalcordtoolbox/reports/assets/_assets/js/bootstrap.min.js /Users/julien/Desktop/sct_course_london20/multi_subject/results/qc/_assets/js
cp /Users/julien/code/sct/spinalcordtoolbox/reports/assets/_assets/js/animation.js /Users/julien/Desktop/sct_course_london20/multi_subject/results/qc/_assets/js
Traceback (most recent call last):
  File "/Users/julien/code/sct/scripts/sct_deepseg_sc.py", line 205, in <module>
    main()
  File "/Users/julien/code/sct/scripts/sct_deepseg_sc.py", line 199, in main
    dataset=qc_dataset, subject=qc_subject, process='sct_deepseg_sc')
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 664, in generate_qc
    angle_line=angle_line
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 545, in add_entry
    layout(qcslice)
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 332, in wrapped_f
    self.qc_report.update_description_file(img.shape)
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 483, in update_description_file
    self._update_html_assets(get_json_data_from_path(path_json))
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 499, in _update_html_assets
    os.makedirs(dest_full_path)
  File "/Users/julien/code/sct/python/envs/venv_sct/lib/python3.6/os.py", line 220, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/Users/julien/Desktop/sct_course_london20/multi_subject/results/qc/_assets/imgs'
~~~

**Comment by cfhammill** [comment]
Would it be sufficient to change:

https://github.com/neuropoly/spinalcordtoolbox/blob/master/spinalcordtoolbox/reports/qc.py#L499

from 

```python
os.makedirs(dest_full_path)
```

to 

```python
os.makedirs(dest_full_path, exist_ok = True)
```

In that case `os.makedirs` ignores the `File exists` exception.  Or are we worried that there is a real corruption risk? 

It's a bit surprising that this occurs because there is a directory exists check immediately above this code. There are a few other instances in the file of `os.makedirs` that could potentially be changed as well.

**Comment by jcohenadad** [comment]
@cfhammill thanks for looking into it. Yes, the suggestion should work but needs to be tested. The problem is that it is a non-systematic bug (happens with coincidence), so in order to make it more "systematic" we might need to set up a specific testing where >100 QC reports are generated at the same time.

**Comment by jcohenadad** [comment]
Probably fixed in #2628
